bigquery_etl/cli/dryrun.py

```diff
@@ -151,4 +167,15 @@ def _sql_file_valid(
             success = False
         return success, sqlfile
 
+    if validate_union_schemas:
+        try:
+            success = result.validate_union_schemas()
+        except Exception:
+            click.echo(
+                f"Failed to validate union schemas in {sqlfile}:\n{traceback.format_exc(limit=0)}",
+                err=True,
+            )
+            success = False
+        return success, sqlfile
```

Comment:
I mimicked the behavior of the existing `--validate-schemas` option above, where if the option is set the normal dryrun validation doesn't get run.  I'm not sure if that was the intended behavior for the `--validate-schemas` option, but I figured I wouldn't change the status quo for that in this PR.

This also means the `--validate-schemas` and `--validate-union-schemas` options are mutually exclusive (with the `--validate-schemas` option taking precedence).

Reply:
I think this check would fit better in `bqetl query validate` since it's validation that doesn't (AFAICT) need to dry run the whole query. The validate command can optionally invoke dry run.  The `validate_schema()` method does seem to do a dry run of the whole query

Reply:
I had assumed `bqetl query` and its subcommands only worked on ETLs, and with this we also want to check views, but looking at the implementation of `bqetl query validate` it seems like it would also process views by default, so I guess that's workable. @scholtzan do you have an opinion on this?

(on a semi-related note, I don't love that `bqetl query validate` defaults to automatically running `bqetl format`, which can mess with your working copy files when running locally unless you remember to specify `--skip-format-sql`)

Reply:
Yeah, slight preference to move it to `bqetl query validate`.